react/week4/github-user-app/src/components/UserContext.js

```diff
@@ -0,0 +1,60 @@
+import React, { useState, createContext } from "react";
+export const UserContext = createContext();
+
+export const UserProvider = ({ children }) => {
+    const [isLoading, setIsLoading] = useState(false);
+    const [users, setUsers] = useState([]);
+    const [userSearch, setUserSearch] = useState("");
+    const [error, setError] = useState("");
+    const [repos, setRepos] = useState([]);
+
+    const fetchUsers = async (url) => {
+        try {
+            const fetchUsers = await fetch(url)
+            const usersData = await fetchUsers.json();
+            console.log(usersData)
+            if (usersData.items) {
+                setIsLoading(false)
+                return setUsers(usersData.items)
+            } else if (usersData.message) {
+                console.log(error)
+                setIsLoading(false)
+                return setError([usersData.message]);
```

Comment:
Why are you setting the error as an array? :)

Reply:
If you know this is an error, you can always `throw new Error(usersData.message)` and then let the `catch` block handle it